bigchaindb/elections/election.py

```diff
@@ -186,49 +191,54 @@ def get_commited_votes(self, bigchain, election_pk=None):
                                                                   election_pk))
         return self.count_votes(election_pk, txns, dict.get)
 
-    def has_concluded(self, bigchain, current_votes=[], height=None):
+    def has_concluded(self, bigchain, current_votes=[]):
         """Check if the election can be concluded or not.
 
-        * Elections can only be concluded if the current validator set
-          is exactly equal to the validator set encoded in the election outputs.
+        * Elections can only be concluded if the validator set has not changed
+          since the election was initiated.
         * Elections can be concluded only if the current votes form a supermajority.
 
         Custom elections may override this function and introduce additional checks.
         """
+        if self.has_validator_set_changed(bigchain):
+            return False
 
         election_pk = self.to_public_key(self.id)
         votes_committed = self.get_commited_votes(bigchain, election_pk)
         votes_current = self.count_votes(election_pk, current_votes)
-        current_validators = self.get_validators(bigchain, height)
 
-        if self.is_same_topology(current_validators, self.outputs):
-            total_votes = sum(current_validators.values())
-            if (votes_committed < (2/3) * total_votes) and \
-                    (votes_committed + votes_current >= (2/3)*total_votes):
-                return True
+        total_votes = sum(output.amount for output in self.outputs)
+        if (votes_committed < (2/3) * total_votes) and \
+                (votes_committed + votes_current >= (2/3)*total_votes):
+            return True
+
         return False
 
     def get_status(self, bigchain):
-        concluded = self.get_election(self.id, bigchain)
-        if concluded:
+        election = self.get_election(self.id, bigchain)
+        if election and election['is_concluded']:
             return self.CONCLUDED
 
+        return self.INCONCLUSIVE if self.has_validator_set_changed(bigchain) else self.ONGOING
+
+    def has_validator_set_changed(self, bigchain):
         latest_change = self.get_validator_change(bigchain)
+        if latest_change is None:
+            return False
+
         latest_change_height = latest_change['height']
-        election_height = bigchain.get_block_containing_tx(self.id)[0]
 
-        if latest_change_height >= election_height:
-            return self.INCONCLUSIVE
-        else:
-            return self.ONGOING
+        election = self.get_election(self.id, bigchain)
+        if not election:
+            return True
+
+        return latest_change_height > election['height']
```

Comment:
The domain in which the above condition with return `false` is as follows,
- `latest_change_height == election['height']`
- `latest_change_height < election['height']`

I understand that the second point will never occur but if we do `return latest_change_height != election['height']` then it would not return `false` for the second point and the logic becomes more transparent.